src/frontend/src/binder/select.rs

```diff
@@ -178,6 +180,14 @@ impl Binder {
 
         // Bind GROUP BY clause.
         self.context.clause = Some(Clause::GroupBy);
+        let number_of_group_keys = select.group_by.len();
+        if number_of_group_keys > MAX_GROUP_KEYS {
```

Comment:
I'm not sure whether this can cover all cases. For example, the columns in `OVER (PARTITION BY <cols>)` may also be used as a group key.

Reply:
With workaround listed below, I think we can remove this, since no more max group keys.

Reply:
Is it possible to exhaustively catch all the cases in frontend? Do you have any suggestions 🙏